solr/core/src/java/org/apache/solr/handler/component/TermVectorComponent.java

```diff
@@ -277,15 +277,15 @@ public void process(ResponseBuilder rb) throws IOException {
       if (null != fields) {
         for (Map.Entry<String, FieldOptions> entry : fieldOptions.entrySet()) {
           final String field = entry.getKey();
-          final Terms vector = reader.getTermVector(docId, field);
+          final Terms vector = reader.termVectors().get(docId).terms(field);
```

Comment:
I think `reader.termVectors()` can be pulled out of the while loop? near line 253